### DISCUSSION TITLE
Phidget 1048 on Artisan showing wrong bean temperature on MacBook Air

### DISCUSSION THREAD

**Discussion by CarlosB2907** [artisan-roaster-scope/artisan#971] [Q&A]
Hi all,

I am a specialty coffee roaster in Milan, ITA, and I've been working with Artisan since 2018. Recently I've changed my PC to a MacBookAir (macOS Big Sur) and unfortunately after mane trial I can get right the bean temperature. I use a Phidget 1048 which is connected to a Probe PT100 - 3W on a small Giesen W1A. I have download the newest Artisan software version for Mac and the newest driver version for the Phidget. Connecting to my machine the temperature shown is around 26 /28°C. But initiating the flame and going to actual high temperature (like 100 /150° C), Artisan displays not more than 45°C. In other words, apparently it is reading somehow the temperature, but not in the right way.

Any one could help me on that? Like with a steep by step on what I should do?

Many thanks,
Carlos Bitencourt

**Comment by MAKOMO** [comment]
The default Phidget setup of Artisan assume k-type probes. Most likely you have a different type of TCs. Second most used once in roasting are j-type.

You can set the type of the probes per channel in the Phidget tab (menu Config >> Device, 4th tab).

<img width="598" alt="Screenshot 2022-09-28 at 09 24 34" src="https://user-images.githubusercontent.com/13909011/192715205-57d46253-ce79-4f0d-9b91-52f3e1d4e498.png">

**Reply by CarlosB2907** [reply]
Hello Mark, 
Thank you.

I have just change now the probe with a K-type but somehow Artisan reads an off-temperature (showing 300/350°C, coming up and down every 5 seconds). It used to work with that probe previously when I was using Windows. 

Do you have any idea for solving this issue? The drivers have been just updated. Perhaps I have the wrong settings in the Config >> Device as you have mentioned, but I don't know the right ones to make it work.

Many thanks,
Carlos

**Reply by MAKOMO** [reply]
You did change the probe or the Artisan setup!? Maybe your probe is not k-type? Before you mentioned it is a PT100 type. Those cannot be read by a Phidget 1048. 

If you have a Phidget 1048 you need to use "Phidget 1048" as the main device type and maybe the "Phidget 1048 34" as an extra device to access the two other channels.

If everything was running fine under Windows, why don't you just export the Artisan settings from your Windows installation (menu Help >> Export Settings) and import it in Artisan running on your mac (menu Help >> Import Settings)?

**Reply by poundy** [reply]
Also, another suggestion, if you still have the Windows PC, why not reconnect it and prove the system still works, particularly if you've changed things like probes. That will help minimise the amount of changes we're trying to help you understand and make at any one point in time

**Reply by hermetheuscoffee** [reply]
Your probe isn't compatible with the Phidget 1048. A PT100 is an RTD, not a thermocouple. RTDs measure temperature by varying resistance, while thermocouples vary voltage. To use your PT100 with Phidgets, you'd need the Phidget TMP1200_0. The "PT" part of the probe indicates that it's platinum (Pt) and the 100 means it has a resistance of 100 ohms at 0 degrees C.

**Comment by CarlosB2907** [comment]
Dear Marko, dear all,

Many thanks. I have reconnected the probe/roaster to the Windows PC, but
somehow the bean temperature reading is showing me very weird temperatures
(varying every 2 or 3 seconds between 200 and 380°C...). I have done what
Marko suggested from my Windows PC to my Mac and the results on Mac are the
same.

Do you know what it could be? I have some hypotheses:
- Phidgets not on the correct Driver version.. (do you know which version
would be the right one?)
- settings on the Artisan software are not correct (do you know what I
should do to correct set Artisan?)
- Probe is not working well.. I've just ordered a new one from Giesen

Many thanks again!
Carlos

_________________________
*Ing. Carlos Eduardo Bitencourt*
C A F E Z A L s.r.l.
T: +39.347.4636.824

**Reply by MAKOMO** [reply]
> - Phidgets not on the correct Driver version.. (do you know which version would be the right one?)

Unlikely. Always use the latest Phidget driver and after installation use its Phidget Control Panel to update your modules firmware version as documented on the Phidget pages.

> - settings on the Artisan software are not correct (do you know what I should do to correct set Artisan?)

Impossible. No combination will produce such swings. Note that Artisan is just displaying the data it receives.

> - Probe is not working well.. I've just ordered a new one from Giesen

Most likely it is the probe. If it its internal insolation breaks the sensor element can (on vibrations) come in contact with the probe shield which is at the ground level of your roasting machine. This will produce such spikes you are observing.

**Comment by CarlosB2907** [comment]
Hello all,
I've connected the new probe in the Giesen W1A, but somehow I still can't
connect/activate the Giesen with Artisan. The probe is correctly working as
well as the LAN cable to my PC, but I still can't see the temperature on
Artisan, nor change the roasting parameters (flame %, air flow, etc..).
Could you please help me in connecting it?
I am not sure the IP Address I'm using is correct (photo attached), or
other parameters. It is a roaster from March 2018 - W1A
thanks once again for your support.
Regards,
Carlos

**Reply by poundy** [reply]
No picture is attached. Please don't try to use Email, only come here to github and attach the image here.

**Comment by CarlosB2907** [comment]
<img width="1279" alt="Screenshot 2022-10-10 at 17 55 54" src="https://user-images.githubusercontent.com/114167077/194907213-6c2c2c07-390a-4c07-90fc-d12abdb7b323.png">

**Reply by CarlosB2907** [reply]
Thanks,
Attached the picture. I am not sure the IP address is correct.  It is a roaster from March 2018 - Giesen W1A
Any idea on what I am doing wrong?

**Comment by CarlosB2907** [comment]
Which is the right one to chose?
Thank you
<img width="380" alt="Screenshot 2022-10-10 at 18 06 44" src="https://user-images.githubusercontent.com/114167077/194909193-a1c90689-2699-489a-a0ad-8e6004bad612.png">

**Comment by MAKOMO** [comment]
Start with the first entry: WxA.

Did you connect your roaster directly via Ethernet wire to your computer? Did you setup a manual network configuration on your PC on the same network as your roaster, just using a different IP address as described [here](https://artisan-scope.org/machines/giesen/)?

**Comment by CarlosB2907** [comment]
Many thanks.
Once the new probe arrives, I will try it.
Regards
_________________________
*Ing. Carlos Eduardo Bitencourt*
C A F E Z A L s.r.l.
T: +39.347.4636.824

**Comment by CarlosB2907** [comment]
Hello all, hello Mark!
I opted today my Artisan and I see this picture..:
<img width="1432" alt="Screenshot 2023-05-16 at 15 13 01" src="https://github.com/artisan-roaster-scope/artisan/assets/114167077/2e80a5e2-3335-42d4-8003-4346a3d56e4a">

Do you know what should I do to see back the curves on artisan, and not this blank page?

Thanks!!!

**Reply by MAKOMO** [reply]
Carlos, I have no idea. Never saw that no chart at all is drawn. Could you please attach your artisan-settings.aset file (you need to rename it to artisan-settings.txt to be able to drag-and-drop it here on Github) as well as your artisan.log file (hold the OPTION key while clicking the grey plus icon in the top left corner and an email with that file attached should open. Please again rename this into artisan.log.txt to be able to drag-and-drop it here. I will take a look at those two files and investigate what is going on here. Thanks!

**Comment by CarlosB2907** [comment]
[artisan-settings_v9 copy.txt](https://github.com/artisan-roaster-scope/artisan/files/11489854/artisan-settings_v9.copy.txt)

**Comment by CarlosB2907** [comment]
Hello Marko,

Thanks. I've sent you one file but I'm not sure what you mean with Log
file..

_________________________
*Ing. Carlos Eduardo Bitencourt*
C A F E Z A L s.r.l.
T: +39.347.4636.824

**Comment by CarlosB2907** [comment]
[artisan.txt](https://github.com/artisan-roaster-scope/artisan/files/11490004/artisan.txt)
[artisan_device.txt](https://github.com/artisan-roaster-scope/artisan/files/11490008/artisan_device.txt)

**Comment by CarlosB2907** [comment]
Done
_________________________
*Ing. Carlos Eduardo Bitencourt*
C A F E Z A L s.r.l.
T: +39.347.4636.824

**Comment by MAKOMO** [comment]
Ah you managed! I will investigate....

**Comment by MAKOMO** [comment]
There seems to be a problem with your fonts. Are you on Windows or MacOS?

**Comment by CarlosB2907** [comment]
Mac

**Reply by MAKOMO** [reply]
Hm. The settings you attached were written by a Windows 10 system. Are you sure you exported them from this Artisan instance running on your mac? With the settings you provided I cannot reproduce the issue.

Is there a specific reason why you start Artisan from its install volume. That could be the reason for your issue. What happens if you copy the Artisan.app to your Macs Application folder and start it from there?

**Comment by CarlosB2907** [comment]
As info: I am working with a Dutch company called Dutch Master Roasters, and we are their distributors in Italy. I know they are working in the final tuning to make available they roaster to Artisan,

**Reply by MAKOMO** [reply]
Carlos,

Open the Terminal.app and enter the following commands at the prompt (without the hash symbol)

```
# cd ~/.matplotlib
# rm *.json
# rm fontList.cache
```

This clears the font cache of the graphic lib underlying Artisan and should get you running again.

**Comment by CarlosB2907** [comment]
Now it works!
Nevertheless, somehow, I can't connect with Artisan my Dutch Master Roaster machine.
As previously mentioned, I am a local dealer in Italy for them, and I know they just finished the tests for running the machine on Artisan. They told me, their machine is working now on Artisan.

Since they are very new with Artisan, could you help me to identify what should I do to connect it on the machine?
All the cables are connect in my Mac, the software is working, but somehow Artisan still does not identify the machine.
Any idea?
Thanks
![Uploading Screenshot 2023-05-17 at 09.29.15.png…]()

**Reply by MAKOMO** [reply]
Dutch Master Roaster is not (yet) supporting the Artisan project, thus we can not support their machines. Sorry.

**Comment by CarlosB2907** [comment]
Hello MArko, hello all,

I'm connecting back a Giesen W1A automatic to Artisan, but somehow I can "link" it to artisan... see the picture attached.. Unfortunately that doesn't work.
![Screenshot 2023-05-24 at 15 09 55](https://github.com/artisan-roaster-scope/artisan/assets/114167077/8ea8a849-c639-44e6-bd4d-f42069020242)


Do you guys know what should I do? Perhaps is a special IP code I must use... ?

Attached a few pictures of my machine and probe.
![Screenshot 2023-05-24 at 14 51 47](https://github.com/artisan-roaster-scope/artisan/assets/114167077/1c82c5d3-143e-4fd7-8932-28716d2ce68b)
![IMG_1817 2](https://github.com/artisan-roaster-scope/artisan/assets/114167077/357141ee-fa6e-4f17-a1d2-09c2cb330034)

![IMG_1818](https://github.com/artisan-roaster-scope/artisan/assets/114167077/efbad529-c780-41b7-af40-82084653d01a)


Thanks,
Carlos

**Reply by poundy** [reply]
Did you read the setup info in https://artisan-scope.org/machines/giesen/ ? Yes, you need to be connected to an Ethernet cable, and you will (likely) need a specific TCP/IP address. Assuming your machine has the appropriate infrastructure (since we don't know if it's a current or older machine that may not have had the required kit - but the ethernet cable in the bottom of the last picture looks promising)

**Comment by CarlosB2907** [comment]
Hello Poundy,
Thanks for you support. I've tried to do that, but somehow does not work.. 
My Giesen is a early 2018 W1A with touch screen. I have added this as IP, which doesn't not work..:
<img width="601" alt="Screenshot 2023-05-25 at 18 46 20" src="https://github.com/artisan-roaster-scope/artisan/assets/114167077/71fd9119-2c89-4830-a95c-f40e34864298">

Could you please help me step by step on what I should do?

**Reply by poundy** [reply]
1) from the guide I linked earlier, and given your date range you quote, I'd assume you fell into this category: 
Default IP Addresses Giesen Roasters before 2/2018 with touch panel:
Model | IP-Roaster | IP-PC (ex)
-- | -- | --
W1 | 192.168.2.180 | 192.168.2.42
So use that as your laptop/computer's IP address not the one from the post 2/2018.
2) check you get a link light when you connect your device to the Giesen. If you don't, it's never going to work, the fundamental electrical connections for the ethernet port (probably the end on your roaster) are not working.
3) I'd ordinarily try pinging the devices address but you can't be sure that the roaster will respond, but worth trying
4) you might try a tool that scans the network for IP addresses that respond. NMAP or similar, you will have to find appropriate details. 
5) perhaps the best thing to do is ring Giesen and confirm what IP address you are meant to use and how to connect.

**Reply by MAKOMO** [reply]
Your computer (configured in your Macs network control panel) needs of course to be set to an IP different than that of the roaster on the same network. Try something like 192.168.2.X with X different from 42. Specify the IP of your roasting machine in Artisan (menu Config >> Ports, MODBUS tab) as 192.168.2.42. Note the IP of your roasting machine might be different as this might be configurable. There is a way to get this touch display to reveal the machines IP address. Use that one and give your computer an IP different as the roasters one, but on the same network (thus just the last number should be to be different).

**Comment by CarlosB2907** [comment]
Thanks.
And which of these should I select?
<img width="340" alt="Screenshot 2023-05-26 at 14 31 01" src="https://github.com/artisan-roaster-scope/artisan/assets/114167077/f0cbdffc-4694-48ef-a78d-2d40083092fa">

**Reply by poundy** [reply]
the one that matches *your* actual roaster? "x" is the size/weight measure. Sorry, you're going to need to figure out which roaster you have on your own I suspect

**Comment by MAKOMO** [comment]
Those setups with an + work only with the Cropster unlock code available for purchase from Giesen. With this the API delivers ET and BT with one decimal instead of zero. 

The setups with the IR or ENV in the name add channels to read in those optional sensors available on some machines by Giesen.

All documented on the corresponding [Supported Machines page](https://artisan-scope.org/machines/giesen/)!
